igraph: 0.10.6 -> 0.10.7

dotlambda opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

dotlambda committed:
igraph: 0.10.6 -> 0.10.7

Diff: https://github.com/igraph/igraph/compare/0.10.6...0.10.7

Changelog: https://github.com/igraph/igraph/blob/0.10.7/CHANGELOG.md

dotlambda committed:
python310Packages.igraph: 0.10.6 -> 0.10.7

Diff: https://github.com/igraph/python-igraph/compare/refs/tags/0.10.6...0.10.7

Changelog: https://github.com/igraph/python-igraph/blob/refs/tags/0.10.7/CHANGELOG.md

dotlambda committed:
igraph: add passthru.tests

szhorvat commented:
Just a note that python-igraph 0.10.8 is out now.

python-igraph 0.10.8 is the counterpart of igraph 0.10.7.  The different versions are not a mistake.

You can consider including python-igraph 0.10.8 in this PR (past PRs updated both the C library and Python package).

dotlambda committed:
python310Packages.igraph: 0.10.7 -> 0.10.8

Diff: https://github.com/igraph/python-igraph/compare/refs/tags/0.10.7...0.10.8

Changelog: https://github.com/igraph/python-igraph/blob/refs/tags/0.10.8/CHANGELOG.md

dotlambda commented:
Result of `nixpkgs-review pr 253964` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>python310Packages.kmapper</li>
    <li>python310Packages.kmapper.dist</li>
    <li>python311Packages.kmapper</li>
    <li>python311Packages.kmapper.dist</li>
  </ul>
</details>
<details>
  <summary>35 packages built:</summary>
  <ul>
    <li>checkov</li>
    <li>checkov.dist</li>
    <li>igraph</li>
    <li>igraph.dev</li>
    <li>igraph.doc</li>
    <li>python310Packages.explorerscript</li>
    <li>python310Packages.explorerscript.dist</li>
    <li>python310Packages.igraph</li>
    <li>python310Packages.igraph.dist</li>
    <li>python310Packages.leidenalg</li>
    <li>python310Packages.leidenalg.dist</li>
    <li>python310Packages.skytemple-dtef</li>
    <li>python310Packages.skytemple-dtef.dist</li>
    <li>python310Packages.skytemple-files</li>
    <li>python310Packages.skytemple-files.dist</li>
    <li>python310Packages.skytemple-ssb-debugger</li>
    <li>python310Packages.skytemple-ssb-debugger.dist</li>
    <li>python310Packages.textnets</li>
    <li>python310Packages.textnets.dist</li>
    <li>python311Packages.explorerscript</li>
    <li>python311Packages.explorerscript.dist</li>
    <li>python311Packages.igraph</li>
    <li>python311Packages.igraph.dist</li>
    <li>python311Packages.leidenalg</li>
    <li>python311Packages.leidenalg.dist</li>
    <li>python311Packages.skytemple-dtef</li>
    <li>python311Packages.skytemple-dtef.dist</li>
    <li>python311Packages.skytemple-files</li>
    <li>python311Packages.skytemple-files.dist</li>
    <li>python311Packages.skytemple-ssb-debugger</li>
    <li>python311Packages.skytemple-ssb-debugger.dist</li>
    <li>python311Packages.textnets</li>
    <li>python311Packages.textnets.dist</li>
    <li>skytemple</li>
    <li>skytemple.dist</li>
  </ul>
</details>
